pgbouncer/docker-entrypoint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+set -ex
+
+cleanup() {
+  echo running cleanup
+  kill -INT "$main_proc"
+  # The signal interrupted our `wait`, so lets do it again
+  wait "$main_proc"
+  echo exited with "$?"
+}
+# SIGTERM - Immediate shutdown. Same as issuing SHUTDOWN on the console.
+# but we would like to catch such signal and replace with SIGINT (Safe shutdown.)
+trap "cleanup" TERM
+
+exec "$@" &
```

Comment:
Does backgrounding exec actually work? I'd expect this to replace the bash process still.

```suggestion
# Run the real command in background so we can still respond to signals.
"$@" &
```

Reply:
yes, check my example

Reply:
```
docker run p
Waiting for it to complete
+ trap cleanup TERM
+ main_proc=7
+ echo 'Waiting for it to complete'
+ wait
+ exec pgbouncer /etc/pgbouncer/pgbouncer.ini
2023-01-19 14:26:52.968 UTC [7] LOG kernel file descriptor limit: 1048576 (hard: 1048576); max_client_conn: 100, max expected fd use: 112
2023-01-19 14:26:52.969 UTC [7] LOG listening on 0.0.0.0:6432
2023-01-19 14:26:52.969 UTC [7] LOG listening on unix:/tmp/.s.PGSQL.6432
2023-01-19 14:26:52.969 UTC [7] LOG process up: PgBouncer 1.17.0, libevent 2.1.12-stable (epoll), adns: c-ares 1.18.1, tls: OpenSSL 1.1.1s  1 Nov 2022
running cleanup
+ cleanup
+ echo running cleanup
+ kill -INT 7
+ wait 7
2023-01-19 14:27:32.598 UTC [7] LOG got SIGINT, shutting down
2023-01-19 14:27:32.614 UTC [7] LOG server connections dropped, exiting
exited with 0
+ echo exited with 0
```

Reply:
@ashb FYI: it works with `exec` and without.